wallet/wallet.c

```diff
@@ -951,6 +974,29 @@ u64 wallet_get_channel_dbid(struct wallet *wallet)
 	return ++wallet->max_channel_dbid;
 }
 
+/* When we receive the remote announcement message, we will also call this function */
+void wallet_announcement_save(struct wallet *w, struct channel *chan)
+{
+	sqlite3_stmt *stmt;
+
+	stmt = db_prepare(w->db, "UPDATE channels SET"
+			  "  remote_ann_node_sig=?,"
+			  "  remote_ann_bitcoin_sig=?"
+			  " WHERE id=?");
+
+	if ((chan->channel_flags & CHANNEL_FLAGS_ANNOUNCE_CHANNEL) &&
+				   chan->remote_ann_node_sig &&
+				   chan->remote_ann_bitcoin_sig) {
+		sqlite3_bind_signature(stmt, 1, chan->remote_ann_node_sig);
+		sqlite3_bind_signature(stmt, 2, chan->remote_ann_bitcoin_sig);
+	} else {
+		sqlite3_bind_null(stmt, 1);
+		sqlite3_bind_null(stmt, 2);
```

Comment:
Arguably we shouldn't be calling `wallet_announcement_save` at all when these are `NULL`.

Reply:
another "overcheck"